sql/core/src/test/scala/org/apache/spark/sql/execution/vectorized/ColumnarBatchSuite.scala

```diff
@@ -1261,4 +1269,38 @@ class ColumnarBatchSuite extends SparkFunSuite {
     batch.close()
     allocator.close()
   }
+
+  testVector("getDecimal should return null for null slot", 4, DecimalType.IntDecimal) {
```

Comment:
shall we make it a normal test case for decimal type? we can follow the other tests, e.g. create a decimal array, and check the value of column vector at the same index.